ADLStream/evaluation/base_evaluator.py

```diff
@@ -1,175 +1,227 @@
-        - Save metrics in `self.metric_history`.
-        - Remove already evaluated data (`y_eval`, `o_eval` and `x_eval`) to keep memory
-          free.
-        - Return new computed accuracy and count of number of instances evaluated.
-
-    Examples:
-    ```python
-        class MinimalEvaluator(BaseEvaluator):
-
-            def __init__(self, metric='kappa', **kwargs):
-                self.metric = metric
-                super().__init__(**kwargs)
-
-            def evaluate(self):
-                new_results = []
-                instances = []
-                current_instance = len(self.metric_history)
-
-                while self.y_eval and self.o_eval:
-                    # Get metric
-                    new_metric = metrics.evaluate(
-                        self.metric,
-                        self.y_eval[0]
-                        self.o_eval[0]
-                    )
-
-                    # Save metric
-                    self.metric_history.append(new_metric)
-
-                    # Remove evaluated data
-                    self.y_eval = self.y_eval[1:]
-                    self.o_eval = self.o_eval[1:]
-                    self.x_eval = self.x_eval[1:]
-
-                    # Add number of instances evaluated
-                    current_instance += 1
-                    instances.append(current_instance)
-
-                retun new_results, instances
-    ```
-
-    Arguments:
-        results_file (str, optional): Name of the csv file where to write results.
-            If None, no csv file is created.
-            Defaults to "ADLStream.csv".
-        dataset_name (str, optional): Name of the data to validate.
-            Defaults to None.
-        show_plot (bool, optional): Whether to plot the evolution of the metric.
-            Defaults to True.
-        plot_file (str, optional): Name of the plot image file.
-            If None, no image is saved.
-            Defaults to None.
-        ylabel (str, optional): y-axis label of the evolution plot.
-            Defaults to "".
-    """
-
-    def __init__(
-        self,
-        results_file="ADLStream.csv",
-        dataset_name=None,
-        show_plot=True,
-        plot_file=None,
-        ylabel="",
-    ):
-        self.results_file = results_file
-        self.dataset_name = dataset_name
-        self.show_plot = show_plot
-        self.plot_file = plot_file
-        self.ylabel = ylabel
-
-        self.x_eval = []
-        self.y_eval = []
-        self.o_eval = []
-        self.metric_history = []
-
-        self._create_results_file()
-
-        if self.show_plot or self.plot_file is not None:
-            self.visualizer = EvaluationVisualizer(self.dataset_name, self.ylabel)
-
-    def _create_results_file(self):
-        if self.results_file is not None:
-            with open(self.results_file, "w") as f:
-                f.write("timestamp,instances,metric\n")
-
-    def start(self):
-        self.visualizer.start()
```

Comment:
You have overridden the previous work. Please make sure you only commit your work and do not modify anything else.